smarts/env/gymnasium/hiway_env_v1.py

```diff
@@ -288,27 +311,38 @@ def step(
 
         assert all("score" in v for v in info.values())
 
-        if self._observations_formatter.observation_options == ObservationOptions.full:
+        if self._environment_return_mode == EnvReturnMode.environment:
             return (
                 self._observations_formatter.format(observations),
                 sum(r for r in rewards.values()),
                 dones["__all__"],
                 dones["__all__"],
                 info,
             )
-        elif self._observations_formatter.observation_options in (
-            ObservationOptions.multi_agent,
-            ObservationOptions.unformatted,
-        ):
-            return (
-                self._observations_formatter.format(observations),
-                rewards,
-                dones,
-                dones,
-                info,
-            )
+        elif self._environment_return_mode == EnvReturnMode.per_agent:
+            observations = self._observations_formatter.format(observations)
+            if (
+                self._observations_formatter.observation_options
+                == ObservationOptions.full
+            ):
+                dones = {**{id_: False for id_ in observations}, **dones}
+                return (
+                    observations,
+                    {**{id_: 0 for id_ in observations}, **rewards},
```

Comment:
In `ObservationsOptions.full`, users can determine whether an agent is active from its `obs.active` value. I was just wondering, should we additionally use `np.nan` (where type(np.nan)=<class 'float'>) instead of  `0` (which is a valid value) for rewards in inactive agents? For example:

```suggestion
                    {**{id_: np.nan for id_ in observations}, **rewards},
```